cmd/promote/git/service_repo.go

```diff
@@ -2,15 +2,14 @@ package git
 
 import (
 	"fmt"
-	"io/ioutil"
 	"os"
 	"strings"
 
 	"github.com/openshift/osdctl/cmd/promote/iexec"
 )
 
-func CheckoutAndCompareGitHash(gitExecutor iexec.IExec, gitURL, gitHash, currentGitHash string) (string, string, error) {
-	tempDir, err := ioutil.TempDir("", "")
+func CheckoutAndCompareGitHash(gitExecutor iexec.IExec, gitURL, gitHash, currentGitHash string, serviceFullPath ...string) (string, string, error) {
```

Comment:
Why is serviceFullPath variadic - it seems we only really handle the case when it's passed in and then always use the first element.

Couldn't it just be a simple string?